sale_order_carrier_auto_assign/models/sale_order.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_confirm(self):
+        for rec in self:
+            if not rec.company_id.carrier_auto_assign:
+                continue
+            rec._add_delivery_carrier_on_confirmation()
+        return super().action_confirm()
+
+    def _add_delivery_carrier_on_confirmation(self):
+        """Automatically add delivery.carrier on sale order confirmation"""
+        for order in self:
+            if order.delivery_set:
+                continue
+            delivery_wiz_action = order.action_open_delivery_wizard()
+            delivery_wiz_context = delivery_wiz_action.get("context", {})
+            if not delivery_wiz_context.get("default_carrier_id"):
+                continue
+            delivery_wiz = (
+                self.env[delivery_wiz_action.get("res_model")]
+                .with_context(**delivery_wiz_context)
+                .create({})
+            )
+            delivery_wiz._get_delivery_rate()
```

Comment:
do we really need to call this function, I checked and found that it is mainly called in `onchange` functions and did not see it appear in any test in odoo codebase. I tried removing it and found that the module still works as expected and I guess that just calling `button_confirm` is enough, since in `button_confirm` there is a call to `set_delivery_line` function. I'm not sure if it's completely correct because I might miss something, do you think so?

Reply:
I think you are right

Reply:
thanks for taking the time, maybe it was worth appearing on that line.